studio/src/utils/otel-helpers.ts

```diff
@@ -28,19 +28,19 @@ const isErrorEvent = (event: OtelEvent) => {
   return isExceptionEvent(event) || isErrorLogEvent(event);
 };
 
-export const hasErrorEvent = (span: OtelSpan) => {
+export const hasErrorEvent = (span: Pick<OtelSpan, "events">) => {
   return span.events?.some(isErrorEvent);
 };
 
-export const getErrorEvents = (span: OtelSpan) => {
+export const getErrorEvents = (span: Pick<OtelSpan, "events">) => {
   return span.events?.filter(isErrorEvent) ?? [];
 };
 
-export function isFpxRequestSpan(span: OtelSpan) {
+export function isFpxRequestSpan(span: Pick<OtelSpan, "name" | "kind">) {
```

Comment:
not sure i like the use of `Pick` here to be honest. these functions are discriminators on the OtelSpan type, and so i think they should only work on data that conform to that type.

imo a plain-old javascipt object with "kind" and "name" attrs would return true from this, which i don't think is expected behavior